tools/component-builder/css/lib/varUtils.js

```diff
@@ -0,0 +1,180 @@
+governing permissions and limitations under the License.
+*/
+
+const gulp = require('gulp');
+const postcss = require('postcss');
+const concat = require('gulp-concat');
+const through = require('through2');
+const fsp = require('fs').promises;
+const logger = require('gulplog');
+const path = require('path');
+
+function getVarsFromCSS(css) {
+  let variableList = [];
+  let root = postcss.parse(css);
+
+  root.walkRules((rule, ruleIndex) => {
+    rule.walkDecls((decl) => {
+      let matches = decl.value.match(/var\(.*?\)/g);
+      if (matches) {
+        matches.forEach(function(match) {
+          let varName = match.replace(/var\((--[\w\-]+),?.*?\)/, '$1').trim();
+          if (variableList.indexOf(varName) === -1) {
+            variableList.push(varName);
+          }
+        });
+      }
+    });
+  });
+  return variableList;
+}
+
+function getVarValues(css) {
+  let root = postcss.parse(css);
+  let variables = {};
+
+  root.walkRules((rule, ruleIndex) => {
+    rule.walkDecls((decl) => {
+      variables[decl.prop] = decl.value;
+    });
+  });
+
+  return variables;
+}
+
+function getClassNames(contents, pkgName) {
+  let root = postcss.parse(contents);
+  let classNames = [];
+
+  function addClassname(className) {
+    if (classNames.indexOf(className) === -1) {
+      classNames.push(className);
+    }
+  }
+
+  let result = root.walkRules((rule, ruleIndex) => {
+    let selector = rule.selectors[0];
+
+    if (pkgName === 'page') {
+      className = '';
+      return 'false';
+    }
+
+    rule.selectors.forEach((fullSelector) => {
+      // Skip compound selectors, they may not start with the component itself
+      if (fullSelector.match(/~|\+/)) {
+        return true;
+      }
+
+      let selector = fullSelector.split(' ').shift();
+
+      if (rule.type === 'rule') {
+        let matches = selector.match(/^\.spectrum-[\w]+/);
+        if (matches) {
+          let modSelector = matches[0]
+          addClassname(modSelector);
+        }
+      }
+    });
+  });
+
+  if (classNames.length === 0) {
+    logger.error(`Could not find classNames for ${pkgName}, assuming no classNames`);
+    classNames.push('');
+  }
+
+  logger.debug(`Found classNames ${classNames.join(', ')} for ${pkgName}`);
+
+  return classNames;
+}
+
+function resolveValue(value, vars) {
+  let match = value.match(/var\((.*?),?.*?\)/);
+  if (match) {
+    return vars[matches[0]];
+  }
+  return value;
+}
+
+const varDir = path.join(path.dirname(require.resolve('@spectrum-css/vars')), '..');
+
+async function readDNAVariables(file) {
+  let css = await fsp.readFile(path.join(varDir, 'css', file));
+  let vars = getVarValues(css);
+  return vars;
+}
+
+function getVariableDeclarations(classNames, vars) {
+  let varNames = Object.keys(vars);
+  if (varNames.length) {
+    return `
+${classNames.map((className) => `${className}`).join(',\n')} {
+${varNames.map((varName) => `  ${varName}: ${vars[varName]};`).join('\n')}
+}
+`;
+  }
+
+  return '';
```

Comment:
not sure if that's important but returning an empty string could lead to issues? 
Do you do a length check later on this or just export the empty string?

Reply:
This function needs to return a string value, the output is written to a file. If there are no defined variables, I still want a string, but an empty one.